Core/GDCore/String.cpp

```diff
@@ -403,6 +403,11 @@ String String::LowerCase() const
     return lowerCasedStr;
 }
 
+String String::CapitalizeFirstLetter() const
+{
+  return size() < 1 ? *this : substr(0, 1).UpperCase() + substr(1);
```

Comment:
Safety check: can you double check `substr` implementation to verify that calling `substr(1)` on `gd::String("a")` works (return an empty string, does not crash)

Reply:
I did check the doc of substr and that's what it's supposed to do, but indeed worth double checking

Reply:
With a 1 letter sentence, the character go from lower to upper case without any apparent issue.